File: src/main/scala/im/tox/toktok/app/SimpleDialogs/SimpleColorDialogDesignAdapter.scala
package im.tox.toktok.app.SimpleDialogs

import android.graphics.Color
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.View.OnClickListener
import android.view.{LayoutInflater, View, ViewGroup}
import im.tox.toktok.R


class SimpleColorDialogDesignAdapter(list: List[String]) extends RecyclerView.Adapter[SimpleColorDialogDesignViewHolder] {

  var items = list

  def onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): SimpleColorDialogDesignViewHolder = {
    val itemView: View = LayoutInflater.from(viewGroup.getContext).inflate(R.layout.simple_color_dialog_item, viewGroup, false)
    itemView.setOnClickListener(new OnClickListener {
      override def onClick(v: View): Unit = {
        Log.d("asd", "adsd")
        v.setZ(1000.0f)
      }
    })
    return new SimpleColorDialogDesignViewHolder(itemView)
  }

  def onBindViewHolder(viewHolder: SimpleColorDialogDesignViewHolder, position: Int) = {
    val item: String = items(position)
    Log.d("asdasd", item + "")
    viewHolder.mColor.setBackgroundColor(Color.parseColor(item))
  }

  def getItemCount(): Int = {
    return items.length
  }
}

final class SimpleColorDialogDesignViewHolder(itemView: View) extends RecyclerView.ViewHolder(itemView) with OnClickListener {

  var mColor: View = itemView.findViewById(R.id.simple_color_dialog_item)

  def onClick(v: View) = {
    Log.d("asd", "adsd")
  }


}

File: sources/app/src/main/java/im/tox/toktok/app/simple_dialogs/SimpleColorDialogDesignAdapter.java
package im.tox.toktok.app.simple_dialogs;

import android.graphics.Color;
import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.List;

import im.tox.toktok.R;

final class SimpleColorDialogDesignAdapter extends RecyclerView.Adapter<SimpleColorDialogDesignViewHolder> {

    private static final Logger logger = LoggerFactory.getLogger(SimpleColorDialogDesignAdapter.class);
    private final List<String> items;

    SimpleColorDialogDesignAdapter(List<String> items) {
        this.items = items;
    }

    @Override
    public SimpleColorDialogDesignViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int viewType) {
        CardView itemView = (CardView) LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.simple_color_dialog_item, viewGroup, false);

        itemView.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(@NonNull View v) {
                logger.debug("adsd");
                v.setZ(1000.0f);
            }
        });

        return new SimpleColorDialogDesignViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull SimpleColorDialogDesignViewHolder viewHolder, int position) {
        String item = items.get(position);
        logger.debug(item);
        viewHolder.mColor.setBackgroundColor(Color.parseColor(item));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

}

final class SimpleColorDialogDesignViewHolder extends RecyclerView.ViewHolder implements OnClickListener {

    private static final Logger logger = LoggerFactory.getLogger(SimpleColorDialogDesignViewHolder.class);

    final View mColor;

    SimpleColorDialogDesignViewHolder(@NonNull CardView itemView) {
        super(itemView);
        mColor = itemView.findViewById(R.id.simple_color_dialog_item);
    }

    @Override
    public void onClick(View v) {
        logger.debug("adsd");
    }

}
